api/src/pcapi/routes/backoffice/accounts/blueprint.py

```diff
@@ -1232,3 +1240,52 @@ def get_public_account_history(
     history = sorted(history, key=lambda item: item.actionDate or datetime.datetime.min, reverse=True)
 
     return history
+
+
+@public_accounts_blueprint.route("/<int:user_id>/gdpr-extract", methods=["POST"])
+@atomic()
+@utils.permission_required(perm_models.Permissions.EXTRACT_PUBLIC_ACCOUNT)
+def create_extract_user_gdpr_data(user_id: int) -> utils.BackofficeResponse:
+    if not FeatureToggle.WIP_BENEFICIARY_EXTRACT_TOOL.is_active():
+        return redirect(url_for(".get_public_account", user_id=user_id))
+
+    user = (
+        users_models.User.query.filter(
+            users_models.User.id == user_id,
+            users_models.User.is_beneficiary,
+        )
+        .options(
+            sa.orm.load_only(users_models.User.firstName, users_models.User.lastName),
+            sa.orm.joinedload(users_models.User.gdprUserDataExtract),
+        )
+        .one_or_none()
+    )
+    if not user:
+        raise NotFound()
+
+    if not has_gdpr_extract(user=user):
+        flash("Une extraction de données est déjà en cours pour cet utilisateur.", "error")
+        return redirect(url_for(".get_public_account", user_id=user_id))
+
+    gdpr_data = users_models.GdprUserDataExtract(
+        userId=user_id,
+        authorUserId=current_user.id,
+    )
+    db.session.add(gdpr_data)
+    db.session.flush()
+
+    on_commit(
+        partial(
+            extract_beneficiary_data.delay,
+            payload=gdpr_tasks.ExtractBeneficiaryDataRequest(extract_id=gdpr_data.id),
+        )
+    )
+    flash(f"L'extraction des données de l'utilisateur {user.full_name} a été demandée.", "success")
+
+    return redirect(url_for(".get_public_account", user_id=user_id))
+
+
+def has_gdpr_extract(user: users_models.User) -> bool:
+    if not user.gdprUserDataExtract:
+        return True
+    return any(extract.is_expired for extract in user.gdprUserDataExtract)
```

Comment:
La fonction fait l'exacte opposé de ce qu'elle dit:
```suggestion
def has_gdpr_extract(user: users_models.User) -> bool:
    if not user.gdprUserDataExtract:
        return False
    return any( not (extract.is_expired) for extract in user.gdprUserDataExtract)
```